libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/FeatureImportances.tsx

```diff
@@ -117,21 +116,35 @@ export class FeatureImportancesTab extends React.PureComponent<
           />
         )}
         {this.state.activeFeatureImportancesOption ===
-          FeatureImportancesTabOptions.LocalExplanation && (
-          <IndividualFeatureImportanceView
-            allSelectedItems={this.props.allSelectedItems}
-            features={this.context.modelMetadata.featureNames}
-            jointDataset={this.context.jointDataset}
-            invokeModel={this.props.requestPredictions}
-            selectedWeightVector={this.props.selectedWeightVector}
-            weightOptions={this.props.weightVectorOptions}
-            weightLabels={this.props.weightVectorLabels}
-            onWeightChange={this.props.onWeightVectorChange}
-            selectedCohort={this.context.selectedErrorCohort}
-            modelType={this.props.modelMetadata.modelType}
-            telemetryHook={this.props.telemetryHook}
-          />
-        )}
+          FeatureImportancesTabOptions.LocalExplanation &&
+          !ifEnableLargeData(this.context.dataset) && (
+            <IndividualFeatureImportanceView
+              allSelectedItems={this.props.allSelectedItems}
+              features={this.context.modelMetadata.featureNames}
+              jointDataset={this.context.jointDataset}
+              invokeModel={this.props.requestPredictions}
+              selectedWeightVector={this.props.selectedWeightVector}
+              weightOptions={this.props.weightVectorOptions}
+              weightLabels={this.props.weightVectorLabels}
+              onWeightChange={this.props.onWeightVectorChange}
+              selectedCohort={this.context.selectedErrorCohort}
+              modelType={this.props.modelMetadata.modelType}
+              telemetryHook={this.props.telemetryHook}
+            />
+          )}
+        {this.state.activeFeatureImportancesOption ===
```

Comment:
This isn't really a concern at all, but out of curiosity I'm wondering if it's better/worse or not really any different to do
```
// your solution
{ conditionA && conditionB && <Component1 />}
{ conditionA && !conditionB && <Component2 />}
```
OR
```
{ conditionA && (
    { conditionB ? <Component1 /> : <Component2/> }
)} conditionB && <Component1 />}
{ conditionA && !conditionB && <Component2 />}
```
If you have any thoughts let me know. If not that's fine, just resolve 🙂

Reply:
Thanks Roman, I don't have any thoughts on the second approach. I believe first approach is more readable than the other :D